src/main/java/com/iwaconsolti/school/miguel/service/GradeService.java

```diff
@@ -0,0 +1,110 @@
+package com.iwaconsolti.school.miguel.service;
+
+import com.iwaconsolti.school.miguel.model.Grade;
+import com.iwaconsolti.school.miguel.model.School;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Service
+public class GradeService {
+
+    private final School gerardoInstitute;
+    private final School zetCollege;
+
+    @Value("${school.score.limit:100}")
+    private int limitGrade;
+
+    public GradeService(@Qualifier("GerardoInstitute") School gerardoInstitute, @Qualifier("ZetCollege") School zetCollege){
+        this.gerardoInstitute = gerardoInstitute;
+        this.zetCollege = zetCollege;
+    }
+
+    public Grade createGrade(String schoolName, Grade grade) {
+        if("GerardoInstitute".equalsIgnoreCase(schoolName) && grade.getScore() <= limitGrade){
+            gerardoInstitute.getGrades().put(grade.getId(), grade);
+            log.info("The grade was successfully registered in Gerardo Institute {}",grade);
+        }else if("ZetCollege".equalsIgnoreCase(schoolName) && grade.getScore() <= limitGrade) {
+            zetCollege.getGrades().put(grade.getId(), grade);
+            log.info("The grade was successfully registered in Zet College {}",grade);
+        }
+        return grade;
+    }
+
+    public List<String> getGradesByStudent(String schoolName, Integer id){
+        List<String> studentGrades = new ArrayList<>();
+        if("GerardoInstitute".equalsIgnoreCase(schoolName)){
+            for(Grade grade: gerardoInstitute.getGrades().values()) {
+                if (grade.getStudent().getId() == id) {
+                    studentGrades.add("Name: " + grade.getStudent().getFirstName() + " " + grade.getStudent().getLastName());
+                    studentGrades.add("Age: " + grade.getStudent().getAge());
+                    studentGrades.add("Professor: " + grade.getCourse().getProfessorName());
+                    studentGrades.add("Curse: " + grade.getCourse().getNameCourse());
+                    studentGrades.add("Score: " + grade.getScore());
+                }
+            }
+        }else if("ZetCollege".equalsIgnoreCase(schoolName)) {
+            for(Grade grade: zetCollege.getGrades().values()) {
+                if (grade.getStudent().getId() == id) {
+                    studentGrades.add("Name: " + grade.getStudent().getFirstName() + " " + grade.getStudent().getLastName());
+                    studentGrades.add("Age: " + grade.getStudent().getAge());
+                    studentGrades.add("Professor: " + grade.getCourse().getProfessorName());
+                    studentGrades.add("Curse: " + grade.getCourse().getNameCourse());
+                    studentGrades.add("Score: " + grade.getScore());
+                }
+            }
+        }
+        return studentGrades;
```

Comment:
5.- In here you can return a Message in case there are no grades

Reply:
I understand what you told me but I feel that this is fine since the view is the one that will be in charge of giving the message when the empty list arrives, since that's how I set it up, also if I wanted to return a message I would have to change the type of the method